cirq-core/cirq/devices/superconducting_qubits_noise_properties.py

```diff
@@ -0,0 +1,269 @@
+                if op_id_swapped not in gate_error_dict:
+                    raise ValueError(
+                        f'Operation {op_id} of field {field_name} has errors '
+                        f'but its symmetric id {op_id_swapped} does not.'
+                    )
+
+    @property
+    def qubits(self) -> List['cirq.Qid']:
+        """Qubits for which we have data"""
+        if not self._qubits:
+            self._qubits = sorted(self.t1_ns)
+        return self._qubits
+
+    @classmethod
+    def single_qubit_gates(cls) -> Set[type]:
+        return SINGLE_QUBIT_GATES
+
+    @classmethod
+    def two_qubit_gates(cls) -> Set[type]:
+        return TWO_QUBIT_GATES
+
+    @classmethod
+    def expected_gates(cls) -> Set[type]:
+        return cls.single_qubit_gates() | cls.two_qubit_gates()
+
+    def get_depolarizing_error(self) -> Dict[OpIdentifier, float]:
+        """Returns the portion of Pauli error from depolarization.
+
+        The result of this method is memoized.
+        """
+        if self._depolarizing_error:
+            return self._depolarizing_error
+
+        depol_errors = {}
+        for op_id, p_error in self.gate_pauli_errors.items():
+            gate_type = op_id.gate_type
+            if gate_type in self.single_qubit_gates():
+                if issubclass(gate_type, ops.MeasurementGate):
+                    # Non-measurement error can be ignored on measurement gates.
+                    continue
+                if len(op_id.qubits) != 1:
+                    raise ValueError(
+                        f'Gate {gate_type} only takes one qubit, but {op_id.qubits} were given.'
+                    )
+                time_ns = float(self.gate_times_ns[gate_type])
+                q0 = op_id.qubits[0]
+                # Subtract decoherence error.
+                if q0 in self.t1_ns:
+                    p_error -= decoherence_pauli_error(self.t1_ns[q0], self.tphi_ns[q0], time_ns)
+
+            else:
+                # This must be a 2-qubit gate.
+                if gate_type not in self.two_qubit_gates():
+                    raise ValueError(f'Gate {gate_type} is not in the supported gate list.')
+                if len(op_id.qubits) != 2:
+                    raise ValueError(
+                        f'Gate {gate_type} takes two qubits, but {op_id.qubits} were given.'
+                    )
+                time_ns = float(self.gate_times_ns[gate_type])
+                # Subtract decoherence error.
+                q0, q1 = op_id.qubits
+                if q0 in self.t1_ns:
+                    p_error -= decoherence_pauli_error(self.t1_ns[q0], self.tphi_ns[q0], time_ns)
+                if q1 in self.t1_ns:
+                    p_error -= decoherence_pauli_error(self.t1_ns[q1], self.tphi_ns[q1], time_ns)
+
+            depol_errors[op_id] = p_error
+        # memoization is OK
+        self._depolarizing_error = depol_errors
+        return self._depolarizing_error
```

Comment:
What if the user decides to modify the output dictionary? Would that not lead to unexpected behavior if they use this NoiseProperties later?

Reply:
This is a fair point. IIRC, the intent of this method was purely as a helper for `build_noise_models` - we could make this method private (i.e. `_get_depol_err`) to enforce this?